File: ncarg2d/src/libncarg/spps/getset.f
C
C $Id: getset.f,v 1.8 2008-07-27 00:17:24 haley Exp $
C
C                Copyright (C)  2000
C        University Corporation for Atmospheric Research
C                All Rights Reserved
C
C The use of this Software is governed by a License Agreement.
C
      SUBROUTINE GETSET (VL,VR,VB,VT,WL,WR,WB,WT,LF)
C
C GETSET returns to its caller the current values of the parameters
C defining the mapping from the user system to the fractional system
C (in GKS terminology, the mapping from world coordinates to normalized
C device coordinates).
C
C VL, VR, VB, and VT define the viewport (in the fractional system), WL,
C WR, WB, and WT the window (in the user system), and LF the nature of
C the mapping, according to the following table:
C
C    1  -  x linear, y linear
C    2  -  x linear, y logarithmic
C    3  -  x logarithmic, y linear
C    4  -  x logarithmic, y logarithmic
C
C Declare the common block containing the linear-log and mirror-imaging
C flags.
C
      COMMON /IUTLCM/ LL,MI,MX,MY,IU(96)
      SAVE /IUTLCM/
C
C Define variables to receive the GKS viewport and window.
C
      DIMENSION VP(4),WD(4)
C
C Check for an uncleared prior error.
C
      IF (ICFELL('GETSET - UNCLEARED PRIOR ERROR',1).NE.0) RETURN
C
C Retrieve the number of the current GKS normalization transformation.
C
      CALL GQCNTN (IE,NT)
      IF (IE.NE.0) THEN
        CALL SETER ('GETSET - ERROR EXIT FROM GQCNTN',1,1)
        RETURN
      END IF
C
C Retrieve the definition of that normalization transformation.
C
      CALL GQNT (NT,IE,WD,VP)
      IF (IE.NE.0) THEN
        CALL SETER ('GETSET - ERROR EXIT FROM GQNT',2,1)
        RETURN
      END IF
C
C Pass the viewport definition to the caller.
C
      VL=VP(1)
      VR=VP(2)
      VB=VP(3)
      VT=VP(4)
C
C Pass the linear/log flag and a (possibly modified) window definition
C to the caller.
C
      LF=LL
C
      IF (LL.EQ.1.OR.LL.EQ.2) THEN
        WL=WD(1)
        WR=WD(2)
      ELSE
        WL=10.**WD(1)
        WR=10.**WD(2)
      END IF
C
      IF (MI.GE.3) THEN
        WW=WL
        WL=WR
        WR=WW
      END IF
C
      IF (LL.EQ.1.OR.LL.EQ.3) THEN
        WB=WD(3)
        WT=WD(4)
      ELSE
        WB=10.**WD(3)
        WT=10.**WD(4)
      END IF
C
      IF (MI.EQ.2.OR.MI.GE.4) THEN
        WW=WB
        WB=WT
        WT=WW
      END IF
C
      RETURN
C
      END

File: ncarg2d/man/spps/getset.m
.TH GETSET 3NCARG "March 1993" UNIX "NCAR GRAPHICS"
.na
.nh
.SH NAME
GETSET - returns a set of values which, if used as arguments in a call to
SET, will cause normalization transformation 1, axis linear/log scaling
(internal parameter 'LS'), and axis mirror imaging (internal parameter 'MI')
to be defined in such a way as to duplicate the combined effects of the
current normalization transformation, axis scaling, and axis mirror imaging.
.SH SYNOPSIS
CALL GETSET (VL,VR,VB,VT,WL,WR,WB,WT,LS)
.SH C-BINDING SYNOPSIS
#include <ncarg/ncargC.h>
.sp
void c_getset(float *vl, float *vr, float *vb, float *vt, float *wl, float *wr, float *wb, float *wt, int *lf)
.SH DESCRIPTION 
.IP VL 12
(an output variable of type REAL) defines the left margin of the
viewport in fractional coordinates.
.IP VR 12
(an output variable of type REAL) defines the right margin of the
viewport in fractional coordinates.
.IP VB 12
(an output variable of type REAL) defines the bottom margin of the
viewport in fractional coordinates.
.IP VT 12
(an output variable of type REAL) defines the top margin of the
viewport in fractional coordinates.
.IP WL 12
(an output variable of type REAL) defines the left margin of the
window in user coordinates.
.IP WR 12
(an output variable of type REAL) defines the right margin of the
window in user coordinates.
.IP WB 12
(an output variable of type REAL) defines the bottom margin of the
window in user coordinates.
.IP WT 12
(an output variable of type REAL) defines the top margin of the
window in user coordinates.
.IP LS 12
(an output variable of type INTEGER) defines the linear/log nature of
the mapping, as follows:
.RS
.IP 1 3
linear X, linear Y
.IP 2 3
linear X, log Y
.IP 3 3
log X, linear Y
.IP 4 3
log X, log Y
.RE
.SH C-BINDING DESCRIPTION
The C-binding argument descriptions are the same as the FORTRAN
argument descriptions.
.SH USAGE
Note that in setting the limits of the viewport and the window, the
left margin can have a user coordinate value that is larger than the
user coordinate value at the right
margin.  This means the X axis scale will run from a maximum
value at the left to a minimum value at the right.  Likewise the
bottom value can exceed the top value meaning that the Y axis can
run from a maximum at the bottom to a minimum at the top.  This
reversal of the axes is referred to as "mirror imaging".
It represents a substantial functional
enhancement over what is provided in GKS.  That is why NCAR Graphics
SET calls using fractional and user coordinates are in general preferred
over GKS calls to GSVP and GSWN using normalized device coordinates
and world coordinates.
.SH EXAMPLES
Use the ncargex command to see the following relevant examples: 
arex01, mpex07, mpex09, cpexcc.
.SH ACCESS
To use GETSET or c_getset, load the NCAR Graphics libraries ncarg, ncarg_gks,
and ncarg_c, preferably in that order.
.SH SEE ALSO
Online:
gqnt, set, getusv, setusv, spps, spps_params, ncarg_cbind
.sp
Hardcopy:  
NCAR Graphics Fundamentals, UNIX Version;
User's Guide for NCAR GKS-0A Graphics
.SH COPYRIGHT
Copyright (C) 1987-2009
.br
University Corporation for Atmospheric Research
.br
The use of this Software is governed by a License Agreement.
